### ISSUE TITLE
Adding folders to the home tab and managing score files within it

### ISSUE THREAD

**Issue by Stccl** [musescore/MuseScore#18059]
### Your idea

You should be able to create folders in the Home tab, so that the user can organize it as they see fit. In the case of the section where the scores from the online manager are displayed (which is supposed to come in 4.1), the layout of these folders could even be synced to your account so that it transfers from one computer to another. There could also be an option to transfer an already existing folder organization from the OS file explorer to MuseScore.
Moreover, it could be nice to be able to manage score files directly from the home tab, completely eliminating the need for the OS file explorer. This could take the form of a right click menu, that would allow you to:

- Rename the score;
- Delete the score;
- Open the score file location in the OS browser;
- Save the score to the cloud;
- Publish the score to MuseScore.com;
- Export the score;
- Create a copy of the score;
- Download the score (in the case of a score in the online score manager).

### Problem to be solved

As it is currently, the home tab is an actual mess in my opinion. If you often use MuseScore, you end up filling it with scores quite quickly and it becomes unusable since there's really no way to organize it. For example, if you bring a small modification to an old score, it's bumped up and ends up among the ones you're currently working on, since it sorts the scores by modification date.

I think it's important to be able to organise it, so that in one glance, you know which score is where, hence my suggestion of adding folders. Thanks for coming to my TED talk lol

### Prior art

**My main inspiration here is the DaVinci Resolve project manager. Have some screenshots:**

![image](https://github.com/musescore/MuseScore/assets/50422710/f966902e-3c6e-4aff-aa01-2640461e806e)

**Right click menu on a folder:**

![image](https://github.com/musescore/MuseScore/assets/50422710/b2c3fa9d-9470-4045-979c-076ff04044d0)

**And on a project (equivalent to a score):**

![image](https://github.com/musescore/MuseScore/assets/50422710/8decc93f-b575-4945-b5b2-0db20b8ff4a9)

**Default right click menu, basically when you right click on nothing:**

![image](https://github.com/musescore/MuseScore/assets/50422710/5c292c28-dcc4-49df-ab10-0bdd8bf3feac)

### Additional context

As someone who uses two computers on a daily basis, I couldn't be more grateful that the score manager will soon be introduced to the Home tab. This shows that you care about it and you want it to have more features and usability. Thanks! I hope you'll consider my suggestion seriously.

**Comment by Stccl** [comment]
Here is what has been said about this suggestion on the discord: https://discord.com/channels/818804595450445834/939880479887327302/1120474644974141511

**Comment by mnaylor375** [comment]
Yes, this would be very helpful. I'm up to about 40 scores and it is very cumbersome to find what I want.
I came here to request this feature, so thank you lunaaccordian for your well-presented request, and thank you MuseScore developers for a great program!

**Comment by MarcSabatella** [comment]
While it's certainly possible such a feature might be added someday, meanwhile be to take full advantage of the normal folder organization scheme provided by your comptuer's operating system - you should not have to rely only on the recent scores list.  With proper use of the folders you already have available, it shouldn't be any more cumbersome to access your scores than it would be if similar functionality were also duplicated on the Home screen.

If you have further questions about how to use folders to organize your scores, or any other questions about how to use MuseScore effectively, please ask on the official Support forum at musescore.org.

**Comment by NickHooked** [comment]
I would also find this useful for an existing filesystem-based project structure.

My scores are organized roughly like this:

```text
project/
  songs/
    song-a/scores/*.mscz
    song-b/scores/*.mscz
```

MuseScore Studio 4.7.4 lets me configure `project/songs` as the Scores folder, but it does not recursively discover the scores in those subfolders. As a result, I cannot use the Home tab as a view of the project without flattening the directory structure or maintaining a separate folder of symlinks.

It would be helpful if MuseScore could recursively index the configured Scores folder while preserving and displaying its directory hierarchy. An optional recursive-scan toggle would also work well for users who prefer the current behavior.

Environment: MuseScore Studio 4.7.4 on Arch Linux / Hyprland.
